common/src/IO/ObjParser.cpp

```diff
@@ -0,0 +1,166 @@
+/*
+ Copyright (C) 2010-2017 Kristian Duske
+
+ This file is part of TrenchBroom.
+
+ TrenchBroom is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ TrenchBroom is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with TrenchBroom. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "ObjParser.h"
+
+#include "Logger.h"
+#include "Assets/EntityModel.h"
+#include "Assets/Texture.h"
+#include "IO/FileSystem.h"
+#include "IO/FreeImageTextureReader.h"
+#include "IO/Path.h"
+#include "IO/Quake3ShaderTextureReader.h"
+#include "Renderer/PrimType.h"
+#include "Renderer/TexturedIndexRangeMap.h"
+#include "Renderer/TexturedIndexRangeMapBuilder.h"
+
+#include <vecmath/forward.h>
+#include <kdl/string_utils.h>
+
+#include <functional>
+#include <string>
+
+namespace TrenchBroom {
+    namespace IO {
+
+        ObjVertexRef::ObjVertexRef(const std::string& text) : m_texcoord(0) {
+            auto components = kdl::str_split(text, "/");
+            std::vector<size_t> components_num;
+            for (const std::string & com : components)
+                components_num.push_back((size_t) std::stoul(com));
```

Comment:
use `kdl::str_to_size` instead, but take care of `nullopt`

Reply:
Resolved as of 5993f765ab.